src/data/setoid/partition.lean

```diff
@@ -237,6 +239,20 @@ _ (subtype (@is_partition α)) _ (partial_order.to_preorder _) $ partition.order
 
 end partition
 
+/-- A finite setoid partition furnishes a finpartition -/
+def is_partition.finpartition {c : finset (set α)}
+  (hc : setoid.is_partition (c : set (set α))) : finpartition (set.univ : set α) :=
+{ parts := c,
+  sup_indep := finset.sup_indep_iff_pairwise_disjoint.mpr $ eqv_classes_disjoint hc.2,
+  sup_parts := c.sup_id_set_eq_sUnion.trans hc.sUnion_eq_univ,
+  not_bot_mem := hc.left }
+
+/-- A finpartition gives rise to a setoid partition -/
+theorem finpartition.is_partition_parts (f : finpartition (set.univ : set α)) :
+  setoid.is_partition (f.parts : set (set α)) :=
+⟨f.not_bot_mem, eqv_classes_of_disjoint_union
+  (f.parts.sup_id_set_eq_sUnion.symm.trans f.sup_parts) f.sup_indep.pairwise_disjoint⟩
+
 end setoid
```

Comment:
You need to put this after `end setoid`, like you had it before. Otherwise it ends up named `setoid.finpartition.is_partition_parts`. This is a `namespace` not a section
```suggestion
end setoid

/-- A finpartition gives rise to a setoid partition -/
theorem finpartition.is_partition_parts (f : finpartition (set.univ : set α)) :
  setoid.is_partition (f.parts : set (set α)) :=
⟨f.not_bot_mem, eqv_classes_of_disjoint_union
  (f.parts.sup_id_set_eq_sUnion.symm.trans f.sup_parts) f.sup_indep.pairwise_disjoint⟩
```